### PR TITLE
Create docker-build.yml

### PR DESCRIPTION
**Pull request by szaimen**
For https://github.com/pulsejet/memories/issues/834

### CODE REVIEW

File: .github/workflows/docker-build.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v3 
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
```
Comment (szaimen) [inline .github/workflows/docker-build.yml:31]: you would need to add these as action secrets in the repo settings

Comment (szaimen) [reply .github/workflows/docker-build.yml:31]: Even the free account on docker hub is sufficient btw. You don't need to buy any subscription there.

File: .github/workflows/docker-build.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+      - name: Build container image
+        uses: docker/build-push-action@v5
+        with:
+          push: true
+          platforms: linux/amd64,linux/arm64
+          context: './'
+          no-cache: true
+          file: 'Dockerfile'
+          tags: pulsejet/go-vod:${{ github.event.inputs.tagName }}
```
Comment (szaimen) [inline .github/workflows/docker-build.yml:41]: the name can be changed of course to something else if required

File: .github/workflows/docker-build.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        uses: docker/login-action@v3 
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build container image
+        uses: docker/build-push-action@v5
+        with:
+          push: true
+          platforms: linux/amd64,linux/arm64
```
Comment (pulsejet) [inline .github/workflows/docker-build.yml:37]: Very cool, I didn't know we could build `arm` images with QEMU.

File: .github/workflows/docker-build.yml (hunk 1 above)
Comment (pulsejet) [reply .github/workflows/docker-build.yml:31]: How about using the GitHub container registry instead? It seems to work pretty seamlessly with e.g. `docker-compose`

Comment (szaimen) [reply .github/workflows/docker-build.yml:31]: Aio needs the images to be published on docker hub bwcause update checks do not work with github container registry. However you could of course publish it on both

Comment (pulsejet) [reply .github/workflows/docker-build.yml:31]: Ok I see. Let me set that up then

Comment (szaimen) [reply .github/workflows/docker-build.yml:31]: I guess you only need to create a free account, add a secret in the account settings and add both account and secret to this repo as variable for actions. Then running the workflow should suceed and automatically create the repo on docker hub

**Comment by pulsejet** [comment]
So I just found https://hub.docker.com/r/linuxserver/ffmpeg

This gives me an idea: it might make sense to just inherit from this (or such an) image, not build our own. There's more: it's quite unnecessary to build a go-vod version-specific image when each release of memories ships with the correct version.

Here's what I'm thinking. When you start the go-vod container, you specify the URL to Nextcloud as an ENV variable. A small bash script runs on the go-vod container that fetches the correct go-vod binary from Nextcloud and starts it up. Now this means that when you upgrade Memories, there's no need to rebuild the container. Just restart it and it'll fetch the new binary. Even better: we can just make an HTTP call to the go-vod container from Nextcloud on upgrade to kill and restart it: seamless upgrades.

**Comment by szaimen** [comment]
> This gives me an idea: it might make sense to just inherit from this (or such an) image, not build our own. There's more: it's quite unnecessary to build a go-vod version-specific image when each release of memories ships with the correct version.

makes sense



> Here's what I'm thinking. When you start the go-vod container, you specify the URL to Nextcloud as an ENV variable. A small bash script runs on the go-vod container that fetches the correct go-vod binary from Nextcloud and starts it up. Now this means that when you upgrade Memories, there's no need to rebuild the container. Just restart it and it'll fetch the new binary. Even better: we can just make an HTTP call to the go-vod container from Nextcloud on upgrade to kill and restart it: seamless upgrades.

This reminds me of what we are doing in the notify-push container:
https://github.com/nextcloud/all-in-one/tree/main/Containers/notify-push 
https://github.com/nextcloud/all-in-one/blob/13966948b05bc5c476cef724e486c799229a0501/php/containers.json#L226-L262

**Comment by pulsejet** [comment]
It's up! https://hub.docker.com/r/radialapps/go-vod

I've adjusted the workflow as above (docs still need work). Basically this image is just ffmpeg+drivers+a small bash script. The bash scripts connects to Nextcloud, downloads the `go-vod` binary and runs it. If it realises the binary is too old at some point (say after upgrading Memories), it'll kill itself and restart.

Of course all this only works on master right now, so need to hold off updating docs at / about AIO until v6 is out.

**Comment by szaimen** [comment]
> It's up! https://hub.docker.com/r/radialapps/go-vod

All right, nice! 🎉

> I've adjusted the workflow as above (docs still need work). Basically this image is just ffmpeg+drivers+a small bash script. The bash scripts connects to Nextcloud, downloads the `go-vod` binary and runs it. If it realises the binary is too old at some point (say after upgrading Memories), it'll kill itself and restart.

Cool!

> Of course all this only works on master right now, so need to hold off updating docs at / about AIO until v6 is out.

All right, I guess only this is missing then after the release is out: https://github.com/nextcloud/all-in-one/tree/main/community-containers#how-to-add-containers

Also see https://github.com/pulsejet/memories/issues/834#issuecomment-1745526381

**Pull request merged**